### ISSUE TITLE
Select native site dark theme if available

### ISSUE THREAD

**Issue by moham96** [darkreader/darkreader#1327]
Hi,
Some site already has a dark/black theme to select which usually looks better than the one implemented by darkreader, so it would be nice if darkreader can detect if the website has a dark theme and activate it instead of color inverting.
This can be implemented by having a database of these sites similar to [dark-sites.config](https://github.com/darkreader/darkreader/blob/master/src/config/dark-sites.config)
the database should contain the domain along with how the dark mode can be activate (usually cookie can influence the theme or DOM object that can be clicked.

**Comment by FranklinYu** [comment]
Example:

* https://support.google.com/a/answer/9296621 see the moon button at bottom-right corner
* https://docs.docker.com/install/overview/ see the toggle at right bar

I'm not sure whether this should include sites that only provides dark theme for logged-in users (like Reddit or Twitter) but I prefer not to. Ideas?

**Comment by FranklinYu** [comment]
Actually I hope that websites ultimately use proper CSS for automatic dark mode: #1285 but that's beside the point.

**Comment by FranklinYu** [comment]
Some HTTP endpoint toggling the theme was mentioned in https://github.com/darkreader/darkreader/issues/1837#issuecomment-579782693. So now we have three possible ways to toggle:

1. cookie
2. an endpoint (possibly require some session)
3. clickable DOM object

I assume that DOM object changes more frequently than previous two.

**Comment by Gusted** [comment]
> Some HTTP endpoint toggling the theme was mentioned in [#1837 (comment)](https://github.com/darkreader/darkreader/issues/1837#issuecomment-579782693). So now we have three possible ways to toggle:
> 
> 1. cookie
> 2. an endpoint (possibly require some session)
> 3. clickable DOM object
> 
> I assume that DOM object changes more frequently than previous two.

Or just get the html element and check which color it is (if color = #000 than native dark has been enabled don't need to be darken. Else Darken Native theme hasn't been enabled)

**Comment by KaKi87** [comment]
Well, you still need the cookie or endpoint to enable the native theme, so we might as well use it to detect whether it is enabled or disabled.

**Comment by Gusted** [comment]
But not everyone would want to have native theme. 

> Well, you still need the cookie or endpoint to enable the native theme, so we might as well use it to detect whether it is enabled or disabled.

To detect you can simple just get an element that is static loaded by default and check the styles of it Endpoint or a Cookie will require some debug time to figure out and add them in the code where not everyone is comfortable with for Privacy parts.

**Comment by Gusted** [comment]
Mine reaction was an intetion to add another method of detecting instead of just enabling/disabling it
> > Some HTTP endpoint toggling the theme was mentioned in [#1837 (comment)](https://github.com/darkreader/darkreader/issues/1837#issuecomment-579782693). So now we have three possible ways to toggle:
> > 
> > 1. cookie
> > 2. an endpoint (possibly require some session)
> > 3. clickable DOM object
> > 
> > I assume that DOM object changes more frequently than previous two.
> 
> Or just get the html element and check which color it is (if color = #000 than native dark has been enabled don't need to be darken. Else darken, native theme hasn't been enabled)

**Comment by KaKi87** [comment]
> But not everyone would want to have native theme.

Non-native darking is already what Dark Reader provides today with the dynamic filter.

> Mine reaction was an intetion to add another method of detecting instead of just enabling/disabling it

The cookie/endpoint is just like a light switch : you can turn on and off the lights, but you also can see if the light is currently turned on or off.

**Comment by Yajo** [comment]
FTR github.com dark theme is now automatic:

![image](https://user-images.githubusercontent.com/973709/127048662-c165f992-ff06-4856-92cc-accc654434e6.png)

**Comment by KaKi87** [comment]
Hello there,

You could be interested in my new userscript [Dark Reader dynamic blacklist](https://git.kaki87.net/KaKi87/userscripts/src/branch/master/darkReaderDynamicBlacklist/README.md) :
- it disables Dark Reader on websites using `prefers-color-scheme` ;
- it disables Dark Reader on websites using manual toggle and automatically triggers the toggle when possible ;
- it keeps Dark Reder enabled until login on websites that only provide theming after login ;
- it workarounds subdomain issues.

**Comment by alexanderby** [comment]
The dark theme detection has been implemented. You need to activate it through the UI. I'm closing this as a duplicate of https://github.com/darkreader/darkreader/issues/81

**Comment by KaKi87** [comment]
This feature request is not implemented nor a duplicate because it consists in automatically enabling a website's dark theme when it is actually disabled by default.

Thanks

**Comment by ShalokShalom** [comment]
Does it make sense, to activate this by default now?
